pages/api/playlists/[id].ts

```diff
@@ -0,0 +1,28 @@
+import { getPlaylistById } from '@/lib/spotify';
+import { isBadStatusCode } from '@/lib/utils';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getSession } from 'next-auth/client';
+
+type NextAPIRequestWithId = NextApiRequest & {
+  query: {
+    id: string;
+  };
+};
+
+export default async (req: NextAPIRequestWithId, res: NextApiResponse) => {
+  const session = await getSession({ req });
+
+  const { id } = req.query;
+
+  const playlistIdResponse = await getPlaylistById(session.accessToken, id);
```

Comment:
Let's rename this since it's not an `id` we're getting returned

```suggestion
  const playlistResponse = await getPlaylistById(session.accessToken, id);
```